packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -92,11 +105,13 @@ export function Checkbox({
           </span>
         </span>
 
-        {label != undefined && (
+        {label && (
           <span className={styles.label}>
             <Text>{label}</Text>
           </span>
         )}
+
+        {children && <span className={styles.label}>{children}</span>}
```

Comment:
can also write it this way

```suggestion
        {(label || children) && (
          <span className={styles.label}>
            {label ? <Text>{label}</Text> : children}
          </span>
        )}
```

between the `||` or duplicating `<span className={styles.label}>` i'd choose the `||` since it's much easier to understand than trying to make sure there's no difference between the 2 `span`

Reply:
I still like the way to keep the check of `label` & `children` separate. But I also see your point that we should avoid reusing the `<span>` here @darryltec 
So after chatting with @kingstonfung, Kingston suggested a good way to satisfy both our intentions

```
const labelText = label ? <Text>{label}</Text> : children;

{ labelText && (<span className={styles.label}>{labelText}</span>) }
```

Reply:
oh wait! I can see a problem with the suggestion 👆 , if we defined both label and children, then we will only be able to show one of them 🤔

Reply:
I'm using XOR to prevent both label and children from being defined, because that is undefined behaviour

Reply:
oh okay then!

Reply:
i like that suggestion